src/js/background/assignManager.js

```diff
@@ -114,8 +114,10 @@ const assignManager = {
   },
 
   async handleProxifiedRequest(requestInfo) {
+
+    // The following blocks potentially dangerous requests for privacy that come without a tabId
     if(requestInfo.tabId === -1)
-      return Utils.DEFAULT_PROXY;
+      return Utils.getBogusProxy();
```

Comment:
Instead of generating a bogus proxy, which might have unexpected side-effects, an alternative could be to store the [`requestId`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/proxy/RequestDetails#Type) as "blocked" at this point, register a new `onBeforeRequest` handler, which listens to all request types, and return `cancel: true` in both `onBeforeRequest` handlers if the `requestId` is seen in there.

Reply:
I agree this is a cleaner option in general - two questions, however:
 - Is onBeforeRequest already used somewhere in the extension? If so it might be a good idea to seamlessly integrate it into the call present there
 - Is it guaranteed that the request will be blocked before it takes place if `requestId` is stored after the proxy addListener call?
 - Finally, has it been tested that non-page requests don't actually proxy? Seeing they can be detected in the addListener method itself, perhaps they do get proxified? Might require some elaborate testing with other addons

Reply:
> Is onBeforeRequest already used somewhere in the extension?

It is, but it only listens to `main_frame` requests, and service worker requests are AFAIK not `main_frame`.

> Is it guaranteed that the request will be blocked

Didn't test it myself, but it would only make sense that the proxy listener is called and fully await'ed before `onBeforeRequest` fires, because otherwise the request wouldn't have the correct [`proxyInfo`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/webRequest/onBeforeRequest#details) attached.

> Finally, has it been tested that non-page requests don't actually proxy?

Didn't test that either, but on `about:serviceworkers` the service workers have `userContextId`s attached, so the requests generated from them could've the correct `cookieStoreId`, might be worth checking indeed.